markdown/renderer_heading.go

```diff
@@ -0,0 +1,59 @@
+package markdown
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/mattn/go-runewidth"
+	"github.com/yuin/goldmark/ast"
+)
+
+func (r *render) renderHeading(node *ast.Heading) error {
+	underlineHeading := false
```

Comment:
Why do we need this variable?

Reply:
we underline heading, if `r.mr.underlineHeadings && node.Level <= 2`

we cannot just do `else` here and on the bottom, because even with `r.mr.underlineHeadings` we don't write newline and underlines if `node.Level > 3`

I could possibly simplify this `if` by doing `underlineHeading = r.mr.underlineHeadings && node.Level <= 2`, is it worth it? :D

Reply:
I see - missed this (: